### PR TITLE
Fixing bug and adding registration to the RP

### PR DESCRIPTION
**Pull request by ranishan**
## Description
The change is to resolve following issues:
- We should avoid registering in Full Mode. Due to a bug in the RP the error saying that the VM was already in full mode may not be true. So it is not worth the risk
- We should Run the script to register subscription to sql virtual machine and log it accordingly
- There was a bug in the script which led to script failing if there is only one VM in a Resource Group or subscription. When Get-AzVM returns a single value, typecasting to array does not work.
- There was a bug when writing to log file during subscription related errors overwrote the log file.

## Checklist

- [ ] This pull request was tested on __both of__:
  - [X] PowerShell 5.1 (Windows)
  - [ ] PowerShell 6.x ([Latest PowerShell](https://github.com/PowerShell/PowerShell/releases))
- [X] Scripts do not contain static passwords or other secret tokens.
- [X] All Azure resource identifiers which must be universally unique are guaranteed to be so.

### Testing information

Platform:
windows 10
powershell 5 1 18362 145
az 2 8 0 -1

Incompatible with powershell core 6.1 as mentioned in the script prerequisites

### CODE REVIEW

**Comment by ranishan** [comment]
@sptramer #sign-off

**Comment by ranishan** [comment]
I have tested the script and updated the checkpoints.

File: sql-virtual-machine/register-sql-vms/RegisterSqlVMs.psm1 [hunk 1]
```diff
@@ -712,16 +694,23 @@ function assert-Subscription(
...
     $registration = Get-AzResourceProvider -ProviderNamespace Microsoft.SqlVirtualMachine -ErrorAction SilentlyContinue
     if ((!$registration) -or ($registration[0].RegistrationState -ne 'Registered')) {
-        $errorMessage = "$($Subscription), Subscription $($Subscription) should be registered to 'Microsoft.SqlVirtualMachine'. This registration may take around 5 mins to propogate. Steps to register can be found here: https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-manager-supported-services"
-        Write-Output $errorMessage | Out-File $Global:LogFile
+        # try register subscription to the SqlVirtualMachine
+        $register = Register-AzResourceProvider -ProviderNamespace Microsoft.SqlVirtualMachine -ErrorAction SilentlyContinue
+        if ((!$register) -or ($register.RegistrationState -ne 'Registering')) {
+            $errorMessage = "$($Subscription), Subscription $($Subscription) should be registered to 'Microsoft.SqlVirtualMachine'. Steps to register can be found here: https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-manager-supported-services. This registration may take around 5 mins to propogate."
```
Comment (sdwheeler) [inline sql-virtual-machine/register-sql-vms/RegisterSqlVMs.psm1]: Do not include locales in URLs. This forces the user to English. Change the URL to:

`https://docs.microsoft.com/azure/azure-resource-manager/resource-manager-supported-services`

The Docs platform detects the user's locale based on their browser configuration. This allows the Docs platform to redirect the user to localized content if available in their language.

File: sql-virtual-machine/register-sql-vms/RegisterSqlVMs.psm1 [hunk 2]
```diff
@@ -21,13 +21,14 @@
     - The script needs to be run on Powershell 5.1 (Windows Only) and is incompatible with Powershell 6.x
     - The subscription whose VMs are to be registered, needs to be registered to Microsoft.SqlVirtualMachine resource provider first. This link describes
       how to register to a resource provider: https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-manager-supported-services
-    - The subscriptions provided to the cmdlet should be in the same tenant as the powershell session. You can confirm this by running
-      the command 'Get-AzSubscription' and verifying that the subscription is in the output of the cmdlet.
+    - Run 'Connect-AzAccount' to first connect the powershell session to the azure account.
```
Comment (sdwheeler) [inline sql-virtual-machine/register-sql-vms/RegisterSqlVMs.psm1]: Do not use locales in URLs.

File: sql-virtual-machine/register-sql-vms/RegisterSqlVMs.psm1 (hunk 1 above)
Comment (ranishan) [reply sql-virtual-machine/register-sql-vms/RegisterSqlVMs.psm1]: Thank you Sean. I have updated all occurrences by removing the locale. I have changed few typos as well

**Pull request merged**